### DISCUSSION TITLE
Header and Footer

### DISCUSSION THREAD

**Discussion by jeromeatneotek** [givanz/Vvveb#86] [General]
If I edit the menu, the header or the footer, is there a way for these to automatically update on all pages automatically?

**Comment by givanz** [comment]
What theme are you using? 

Most themes have this option, there are a few that don't have this yet and they will be updated to also include the option.

For Landing theme changing header and footer on any page will apply to all pages.

Homepage has a different header because of hero background image covering the header and changes to home header will not apply to other pages.

In theme html code this is implemented using the `data-v-save-global` attribute.

```html
<nav class="navigation-1 clearfix" data-v-save-global="blank.html,.navigation-1">
```

**Comment by jeromeatneotek** [comment]
I am using the default theme. Ideally I would expect that if I edited the menu, header or footer it would get applied to any pages that reference these. I assume this does not work this way now, would it be possible? I am happy to assist with development, just working my way around the solution currently. Fee free to contact me on discord jerome_digitalrowing

**Comment by givanz** [comment]
Changing the header and footer on any page is applied on all pages except homepage header which has a different layout because of hero.

Menu elements can be edited from `admin > pages > menus > main menu`

Can you try clearing cache? If it still doesn't work let me know what page and elements you edit  and I will try to reproduce the bug.

You can make any area in the templates global by adding the `data-v-save-global="page.html,#selector` attribute.

The first parameter is the html page from where the content is copied and saved and the second is the element selector, for site header it is implemented like this

```html
<nav class="navigation-1 clearfix" data-v-save-global="blank.html,.navigation-1">
```

Sorry, I don't have a discord account.

**Comment by jeromeatneotek** [comment]
Hi Givan

Thank you for coming back to me so quickly, that all sounds great. Do you use any chat tools at all or is email the best way to talk?

I see you also tried using GrapeJS for a while too. I went down this path also, but I like your solution better.

Ideally I would prefer a node solution so that everything is javascript based, would you be interested in this?

Also I noticed you are still using jquery. I have moved all my development away from jquery because it is much slower than vanilla javascript which now provides natively all the functionality that used to require jquery. Would you be interested in assistance with porting the jquery to vanilla javascript?

Best regards
Jerome Howard
Systems Architect |  Neotek Limited

**Comment by givanz** [comment]
Hi Jerome

I mostly use email and github.

I checked many page builder libraries including GrapesJs before writing VvvebJs https://github.com/givanz/VvvebJs many years ago.

The reason I had to write my own was that I needed to edit raw html without any unnecessary and complex parsing that for some strange reason other page builders make reinventing the DOM as javascript objects making everything more complex and memory intensive.

Vvveb CMS is built around VvvebJs and never used a different page builder, having themes use plain html allows more creative freedom and maximum customization within the page builder and also faster development.

I prefer php over node for many reasons but the most important is hosting which for php is more diverse, cheaper and available on a much greater scale this ensures that more people can use the CMS, also deployment and installation is much easier with php.

Rewriting javascript to remove jQuery dependency is not a priority, I know browsers caught up and it is no longer necessary but the library is small and there is no practical disadvantage.

If you want to rewrite the code to remove jQuery code contributions are welcome.

The only reason to remove the dependency is to take advantage of modern browser features and make the code more appealing for developers but for end users there is no advantage.

The main focus is the CMS as a whole and what is needed by the end user, VvvebJs development is shaped by the CMS requirements.

Best regards
Givan

**Reply by jeromeatneotek** [reply]
Thanks for the background, I totally understand and agree with your philosophy.
I noticed that you are advising I download the zip to get updates, however none of these updates are getting checked in to the git repository. Is this just a matter of time?

**Reply by givanz** [reply]
Yes, all changes will be pushed to github, new features that are under development are delayed to allow more testing to commit the code in a more complete form.

All new changes are pushed to latest.zip first and the demo reinstalled to test the changes on the server on a different configuration and also for eventual feedback on new features.
